### DISCUSSION TITLE
Port forwarding not working

### DISCUSSION THREAD

**Discussion by iorran** [prisma/prisma#17452] [Q&A]
### Bug description

I have a Mysql running on AWS, I'm able to connect through intellij or dbveaver by using port forwarding, but, prisma always give me wrong credentials error.

### How to reproduce

Intellij

<img width="741" alt="image" src="https://user-images.githubusercontent.com/5152664/213757930-c2427461-cb4f-410d-a4bc-781d132fd34e.png">

DBMS: MySQL (ver. 5.7.38-log)
Case sensitivity: plain=exact, delimited=exact
Driver: MySQL Connector/J (ver. mysql-connector-java-8.0.25 (Revision: 08be9e9b4cba6aa115f9b27b215887af40b159e0), JDBC4.2)
Ping: 153 ms
SSL: yes

Prisma

<img width="965" alt="image" src="https://user-images.githubusercontent.com/5152664/213758134-63378c50-489e-4215-bea7-712f022771c4.png">

DATABASE_URL="mysql://<user>:<password>@localhost:65432/gesundheit"

### Expected behavior

I want to connect prisma with Port forwarding 

### Prisma information

```prisma
generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider = "mysql"
  url      = env("DATABASE_URL")
}
I remove the tables for readability 

```

```ts
import { INestApplication, Injectable, OnModuleInit } from '@nestjs/common';
import { PrismaClient } from '@prisma/client';

@Injectable()
export class PrismaService extends PrismaClient implements OnModuleInit {
  async onModuleInit() {
    await this.$connect();
  }

  async enableShutdownHooks(app: INestApplication) {
    this.$on('beforeExit', async () => {
      await app.close();
    });
  }
}

```

### Environment & setup

- OS: macOS
- Database: MySQL
- Node.js version: v14.21.2

### Prisma Version

```
prisma                  : 4.9.0
@prisma/client          : 4.7.1
Current platform        : darwin
Query Engine (Node-API) : libquery-engine ceb5c99003b99c9ee2c1d2e618e359c14aef2ea5 (at node_modules/@prisma/engines/libquery_engine-darwin.dylib.node)
Migration Engine        : migration-engine-cli ceb5c99003b99c9ee2c1d2e618e359c14aef2ea5 (at node_modules/@prisma/engines/migration-engine-darwin)
Introspection Engine    : introspection-core ceb5c99003b99c9ee2c1d2e618e359c14aef2ea5 (at node_modules/@prisma/engines/introspection-engine-darwin)
Format Binary           : prisma-fmt ceb5c99003b99c9ee2c1d2e618e359c14aef2ea5 (at node_modules/@prisma/engines/prisma-fmt-darwin)
Format Wasm             : @prisma/prisma-fmt-wasm 4.9.0-42.ceb5c99003b99c9ee2c1d2e618e359c14aef2ea5
Default Engines Hash    : ceb5c99003b99c9ee2c1d2e618e359c14aef2ea5
Studio                  : 0.479.0
```

**Comment by nurul3101** [comment]
Hello @iorran 👋 
Can you double-check if the connection string that you are passing to connect with IntelliJ is the same one that you are passing in the DATABASE_URL environment variable?

> DATABASE_URL="mysql://:@localhost:65432/gesundheit"

Also, Are you able to connect to the database without port forwarding but getting this error only when port forwarding is enabled?

**Reply by iorran** [reply]
@nurul3101 Hi!

So, on the intellij screenshot, the field URL is not the connection string?

Yep, Im not able to use the database without port forwarding

**Reply by nurul3101** [reply]
Just wanted to confirm the connection string is valid as there was an additional colon (:) in the DATABASE_URL, which wasn't in the screenshot.

Can you try adding username and password in the DATABASE_URL and check if you get the same error?

**Reply by iorran** [reply]
Nowadays its been used with user and password... I removed just from the sample
DATABASE_URL="mysql://user:password@localhost:65432/gesundheit"

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
